File: agri-trade/backend/src/main/java/com/agri/trade/entity/Quote.java
package com.agri.trade.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.math.BigDecimal;
import java.util.Date;

@Data
@TableName("agri_trade_quote")
public class Quote {

    @TableId(type = IdType.AUTO)
    private Long id;

    @TableField("quote_no")
    private String quoteNo;

    @TableField("inquiry_id")
    private Long inquiryId;

    @TableField("inquiry_no")
    private String inquiryNo;

    @TableField("product_id")
    private Long productId;

    @TableField("product_name")
    private String productName;

    @TableField("buyer_id")
    private Long buyerId;

    @TableField("buyer_name")
    private String buyerName;

    @TableField("seller_id")
    private Long sellerId;

    @TableField("seller_name")
    private String sellerName;

    @TableField("unit_price")
    private BigDecimal unitPrice;

    @TableField("quantity")
    private BigDecimal quantity;

    @TableField("unit")
    private String unit;

    @TableField("total_amount")
    private BigDecimal totalAmount;

    @TableField("quote_amount")
    private BigDecimal quoteAmount;

    @TableField("freight_amount")
    private BigDecimal freightAmount;

    @TableField("discount_amount")
    private BigDecimal discountAmount;

    @TableField("payment_terms")
    private String payment_terms;

    @TableField("delivery_time")
    private Date deliveryTime;

    @TableField("delivery_deadline")
    private Date deliveryDeadline;

    @TableField("delivery_address")
    private String deliveryAddress;

    @TableField("quality_description")
    private String qualityDescription;

    @TableField("certification_info")
    private String certificationInfo;

    @TableField("packaging_info")
    private String packagingInfo;

    @TableField("valid_until")
    private Date validUntil;

    @TableField("quote_status")
    private String quoteStatus;

    @TableField("buyer_response")
    private String buyerResponse;

    @TableField("response_time")
    private Date responseTime;

    @TableField("order_id")
    private Long orderId;

    @TableField("del_flag")
    private Integer delFlag;

    @TableField("tenant_id")
    private String tenantId;

    @TableField("create_by")
    private String createBy;

    @TableField("create_time")
    private Date createTime;

    @TableField("update_by")
    private String updateBy;

    @TableField("update_time")
    private Date updateTime;

    @TableField("remark")
    private String remark;
}

File: agri-trade/frontend/src/api/quote.ts
import axios from 'axios'

export interface QuoteQuery {
  pageNum: number
  pageSize: number
  quoteNo?: string
  productName?: string
  sellerId?: number
  quoteStatus?: string
}

export interface QuoteVO {
  id: number
  quoteNo: string
  demandId: number
  productId: number
  productName: string
  sellerId: number
  sellerName: string
  unitPrice: number
  quantity: number
  unit: string
  totalAmount: number
  validDays: number
  expireTime: string
  qualityLevel: string
  deliveryTime: string
  deliveryAddress: string
  remark: string
  quoteStatus: string
  createTime: string
  updateTime: string
}

export async function getQuoteList(params: QuoteQuery) {
  const response = await axios.get('/api/trade/quote/list', { params })
  return response.data
}

export async function getQuoteById(id: number) {
  const response = await axios.get(`/api/trade/quote/${id}`)
  return response.data
}

export async function createQuote(data: Partial<QuoteVO>) {
  const response = await axios.post('/api/trade/quote', data)
  return response.data
}

export async function updateQuote(id: number, data: Partial<QuoteVO>) {
  const response = await axios.put(`/api/trade/quote/${id}`, data)
  return response.data
}

export async function deleteQuote(id: number) {
  const response = await axios.delete(`/api/trade/quote/${id}`)
  return response.data
}

export async function acceptQuote(id: number) {
  const response = await axios.post(`/api/trade/quote/${id}/accept`)
  return response.data
}

export async function rejectQuote(id: number, reason: string) {
  const response = await axios.post(`/api/trade/quote/${id}/reject`, { reason })
  return response.data
}